wbooks/src/app/screens/dashboard/index.js

```diff
@@ -0,0 +1,46 @@
+import React, {Component} from 'react';
+import Catalog from './components/Catalog';
+import Search from './components/Search';
+import booksJson from '../../../../resources/books.json';
+
+class Dashboard extends Component {
+  state = {
+    filteredBooks: booksJson,
+    books: booksJson
+  };
+
+  updateFilters = (dropdownFilter, textFilter) => {
+    
+    if(!textFilter && !dropdownFilter) {
+      this.setState({filteredBooks:this.state.books});
+    }
+    else {
+      let allBooks = this.state.books;
+      let newBooks = [];
+
+      for(let i= 0; i < allBooks.length; i++) {
```

Comment:
instead of the foreach, use the function `array.filter`
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter